keel-plugin/src/main/kotlin/com/netflix/spinnaker/keel/annealing/ResourceActuator.kt

```diff
@@ -4,35 +4,41 @@ import com.netflix.spinnaker.keel.api.ApiVersion
 import com.netflix.spinnaker.keel.api.Resource
 import com.netflix.spinnaker.keel.api.ResourceName
 import com.netflix.spinnaker.keel.persistence.ResourceRepository
-import com.netflix.spinnaker.keel.persistence.ResourceState.Diff as Different
+import com.netflix.spinnaker.keel.persistence.ResourceState.Diff
 import com.netflix.spinnaker.keel.persistence.ResourceState.Missing
 import com.netflix.spinnaker.keel.persistence.ResourceState.Ok
 import com.netflix.spinnaker.keel.plugin.ResourceConflict
 import com.netflix.spinnaker.keel.plugin.ResourceHandler
 import com.netflix.spinnaker.keel.plugin.ResourceHandler.ResourceDiff
 import com.netflix.spinnaker.keel.plugin.supporting
+import com.netflix.spinnaker.keel.telemetry.ResourceChecked
 import de.danielbechler.diff.ObjectDifferBuilder
 import de.danielbechler.diff.node.DiffNode
 import org.slf4j.LoggerFactory
+import org.springframework.context.ApplicationEventPublisher
 import org.springframework.stereotype.Component
 
 @Component
 class ResourceActuator(
   private val resourceRepository: ResourceRepository,
-  private val handlers: List<ResourceHandler<*>>
+  private val handlers: List<ResourceHandler<*>>,
+  private val publisher: ApplicationEventPublisher
```

Comment:
I'll rather raise a simple event and handle it out of band that inject `Registry` here.

Reply:
I could get behind this strategy, lets giv'er a go

is the event publisher we are using async and multi-threaded? (I feel like that is maybe a spring config type of thing..) If we are using it for metrics then I think it shouldn't be a single threaded event processing queue as it would kind of suck for metrics to get stuck around some other slow event handler

Reply:
It is following 112e577